src/components/NavBar/NavBarSignInOrAvatar.component.jsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+import { useGlobalContext } from '../../providers/Global/Global.provider';
+
+import Avatar from './SingInAndAvatar/Avatar.component';
+import SignIn from './SingInAndAvatar/SignIn.component';
+
+const SignInOrAvatar = () => {
+  const { globalState } = useGlobalContext();
+
+  const { isAuthenticated } = globalState;
+
+  return <>{isAuthenticated ? <Avatar /> : <SignIn />}</>;
```

Comment:
This ternary could be moved to the NavBar component where you needed it, you already have the `globalState` over there. It will simplified the code and make it more readable, as you don't need to look for this file just to see the conditional rendering.